elastic/kibana
# [Console] Generate autocomplete definitions from ES specification

## Summary

This PR uses the new script to generate autocomplete definitions for Dev Tools Console from the ES specification repo. 

#### Definitions changes
- New property `availability` is added to filter out endpoints that are not available in Serverless
- Some endpoints' query parameters have more details now, for example common query params are now defined in definitions 
```json
"url_params": {
      "error_trace": "__flag__",
      "filter_path": [],
      "human": "__flag__",
      "pretty": "__flag__"
},
```
- Url components in few endpoints are removed, but those were added to overrides files in https://github.com/elastic/kibana/pull/163096
- Documentation links contain `{branch}` instead of `master` (fix for that added in https://github.com/elastic/kibana/pull/159241)

#### Script changes
- The logic for generating `availability` for endpoint has been updated based on the feedback from the Clients team. Details added to the script file. 
- Added a few "safe guards" to the spots in the script where an unexpected type of data might be coming from the ES specification schema

#### Console changes
- Fixed the autocomplete request on Serverless (we might need a proper fix for that, details in https://github.com/elastic/kibana/issues/163318)

Also updates to readme files both in Console and the new script. 
I will remove the old script in a separate PR. 

## Screenshots 
"ILM" autocomplete suggestions displayed on stateful
<img width="583" alt="Screenshot 2023-08-07 at 17 47 48" src="https://github.com/elastic/kibana/assets/6585477/641a48b0-fb1a-4d3b-a8c9-99eab8795510">


"ILM" autocomplete suggestions not displayed on serverless
<img width="572" alt="Screenshot 2023-08-07 at 17 35 16" src="https://github.com/elastic/kibana/assets/6585477/a1ee5468-eb9f-4f52-81d5-c661b06f8ceb">


## How to test
- Start Kibana on stateful (`yarn start`) and check that autocomplete suggestions are working as before (no changes)
- Start Kibana on serverless (`yarn start --serverless`) and check that autocomplete suggestions are not displayed for endpoints blocked/internal on serverless. 

### Checklist

Delete any items that are not applicable to this PR.
